
---
output:
  md_document:
    variant: markdown_github
---

"Vaginal cell herpes infection experiment"


```{r,echo=FALSE,message=FALSE, warning=FALSE}
require(dplyr)
require(lumi)
require(limma)

#READ IN RAW MICROARRAY DATA FINAL REPORT #######
#data sent from LMF

RAW<-"J:\\MacLabUsers\\HLADIK SHARED\\Projects\\Herpes study\\Herpes-Project-1\\vaginal_cell_microarray\\Finalreport_HSV2_HVE MA.txt"

RAW.lumi<-lumiR(RAW,detectionTh = 0.05, na.rm = TRUE,convertNuID = FALSE, dec = '.',
                parseColumnName = FALSE, checkDupId = FALSE,
                QC = TRUE,
                columnNameGrepPattern = list(exprs='AVG_SIGNAL',
                                                  se.exprs='BEAD_STDERR',
                                                       detection='Detection Pval',
                                                       beadNum='Avg_NBEADS'),
                inputAnnotation=TRUE,
                annotationColumn=c('ILMN_GENE', 'ENTREZ_GENE_ID', 'GI', 'ACCESSION', 'SYMBOL', 'PROBE_ID', 'PROBE_START', 'PROBE_SEQUENCE', 'CHROMOSOME', 'PROBE_CHR_ORIENTATION', 'PROBE_COORDINATES'),
                verbose = TRUE)


#Remove sample 3E (column 26) from the data
RAW.lumi<-RAW.lumi[,-26]

```

SOME PLOTS OF NON NORMALIZED DATA:
LMF left sample 3E out of the analysis because it was determined that there was some kind of pipetting error. FH told me that when he looked at the analyzed data, 3E looked very similar to the SD90 data but very different from the other 2 V186.24 replicates (samples 6D and 3D) so he didn't feel comfortable leaving it in the analysis. I am also leaving it out here.

 Density plot, cdf plot, sample relations, boxplot. Looks like 3D and 6D are outliers, both are V186 24hrs.
 
```{r,echo=FALSE,message=FALSE, warning=FALSE}
#density plot
density(RAW.lumi)


#CDF plot: cumulative probability of having <= a certain log2 intensity
plotCDF(RAW.lumi)

#sample relations
plot(RAW.lumi, what='sampleRelation',method="mds")

#boxplot
boxplot(RAW.lumi)


#adding in more complete pData that matches my formatting for the #explant data. 

pData <- read.table("vaginalCellMicroarrayPhenoData.txt",
                       sep="\t",row.names=1, header=TRUE)

pData<- pData[-26,] #remove sample 3E from the pData so it will work nicely with the lumiBatch where I did the same

#create metadata df
metadata<- data.frame(labelDescription = c("PTID","Sampling Time", "Treatment:either a virus strain or Mock."),
                      row.names=c("TissueID","Time","Treatment"))


#combine metadata and pdata into an annotated df
adf<-new("AnnotatedDataFrame",data=pData,varMetadata=metadata)

#create experiment data
experimentData<-new("MIAME",name="Claire Levy",
                    lab="Florian Hladik Lab",title="Vaginal Cell Microarray")


#make a lumiBatch that contains both the raw data we got from
#shared resources AND the phenoData, metadata and experiment data.
complete.RAW.lumi <- new("LumiBatch", exprs = exprs(RAW.lumi),phenoData = adf,
                  experimentData = experimentData,
                  se.exprs = se.exprs(RAW.lumi),
                  detection = detection(RAW.lumi),
                  featureData=featureData(RAW.lumi))

save(complete.RAW.lumi,file = "CELLScomplete.RAW.lumi.Rdata")


###################BACKGROUND CORRECTION ####################
#the data we got from the core had no background correction when we did the explant experiment so I assume this wasn't corrected either.
B.complete.RAW.lumi<-lumiB(complete.RAW.lumi)


#################### VST TRANSFORMATION ######################
#"Stabilizing the expression variance based on
#the bead level expression variance and mean relations"

TB.complete.RAW.lumi <-lumiT (B.complete.RAW.lumi)

################## ROBUST SPLINE NORMALIZATION ################

NTB.complete.RAW.lumi<-lumiN(TB.complete.RAW.lumi,method="rsn")

################# QUALITY CONTROL ############################

QNTB.complete.RAW.lumi <- lumiQ(NTB.complete.RAW.lumi,detectionTh=0.05)

save(QNTB.complete.RAW.lumi, file = "CELLSQNTB.complete.RAW.lumi.Rdata")
```

PLOTS OF NORMALIZED DATA: all data, then just 6D and 3D


```{r,echo=FALSE,message=FALSE, warning=FALSE}

density(QNTB.complete.RAW.lumi)



plot(QNTB.complete.RAW.lumi, what='sampleRelation',method="mds")



boxplot(QNTB.complete.RAW.lumi)


```
FILTERING PROBES BASED ON DETECTION 
 Limma suggests to keep probes that are expressed above bg on 
at least n arrays where n is smallest number of replicates assigned
to any of the treatment combinations.

Our treatment combinations are TissueID/Treatment/Time
We have 3 Tissue IDs x 3 Treatments x 3 time points = 27 so the smallest number
 of replicates possible for any of those 3 arrays is 3

```{r,echo=FALSE,message=FALSE, warning=FALSE}


detectedProbes <- rowSums(detection(QNTB.complete.RAW.lumi)<0.05)>=3

#now extract just those probes that are TRUE from the lumibatch
#meaning there were at least 3 samples with detection p-values
#that were <0.05

CELLSexpressedProbes.lumi <-QNTB.complete.RAW.lumi[detectedProbes,]
```

how many probes did we have before and after filtering?
```{r,echo=FALSE,message=FALSE, warning=FALSE}
dims(QNTB.complete.RAW.lumi)#47323

dims(CELLSexpressedProbes.lumi)#22894
```


How many removed?


```{r,echo=FALSE,message=FALSE, warning=FALSE}

dims(QNTB.complete.RAW.lumi)-dims(CELLSexpressedProbes.lumi)#20192

save(CELLSexpressedProbes.lumi, file="CELLSexpressedProbes.lumi.Rdata")
```



```{r,echo=FALSE,message=FALSE, warning=FALSE}


############### TARGETS AND DESIGN MATRIX ###################
# see section 9.4.1 and 9.4.2 in the limma users guide

targets<-pData(CELLSexpressedProbes.lumi)%>%
  select(TissueID, Treatment,Time)
TissueID<- factor(targets$TissueID)
Treat <-factor(paste(targets$Treatment,targets$Time, sep="."))

design<-model.matrix (~0+Treat+TissueID)
save(design, file="CELLSdesign.Rdata")


```


```{r,echo=FALSE,message=FALSE, warning=FALSE}


####################### FIT MODEL TO PROBES ###################

fit <- lmFit(CELLSexpressedProbes.lumi,design=design)
save(fit, file="CELLSfit.Rdata")

#Now we can make any comparisons
#between the experimental conditions

# If topTable is called and coef has two or more elements,
# then the specified columns will be extracted from fit and
# topTableF called on the result. topTable with coef=NULL is 
# the same as topTableF, unless the fitted model fit has only
# one column.

bothcm<-makeContrasts(
  V186.3vsMock.3 = TreatV186.3-TreatMock.3,
  V186.8vsMock.8 = TreatV186.3-TreatMock.8,
  V186.24vsMock.24 = TreatV186.24-TreatMock.24,
  SD90.3vsMock.3 = TreatSD90.3-TreatMock.3,
  SD90.8vsMock.8 = TreatSD90.8-TreatMock.8,
  SD90.24vsMock.24 = TreatSD90.24-TreatMock.24,
  levels=design
)



#fit the contrasts  
fit2<-contrasts.fit(bothcm, fit=fit)


#compute diff exprsn
fit2 <-eBayes(fit2)

save(fit2, file= "CELLSfit2eBayes.Rdata")
```

How many probes are up and down regulated for each contrast?

```{r,echo=FALSE,message=FALSE, warning=FALSE}

#method=separate is same as doing topTable for all coefs separately
results<-decideTests(fit2,method="separate", adjust.method="BH",
                      p.value=0.05, lfc=0.5)

save(results, file= "CELLSdecideTestsResults.Rdata")

#turn the results matrix into a data frame and make the
#probeID a real column and remove the rownames

resultsDF<-as.data.frame(results)
resultsDF$ProbeID<-rownames(resultsDF)
rownames(resultsDF)<-NULL

#melt the df for easy summarizing
library(reshape2)

resultsDFmelt<-melt(resultsDF, id.vars="ProbeID")
save(resultsDFmelt,file = "resultsDFmelt.Rdata")
#number of up and down regulated probes based on 
#p.val at least 0.05 and lfc at least 0.5

summary<-resultsDFmelt %>%
  group_by(variable)%>%
 summarize(down=sum(value=="-1"),up=sum(value=="1"))
summary
save(summary, file = "CELLSsummary.Rdata")

meltSummary <- melt(summary)

colnames(meltSummary)<-c("Contrast","Direction","Probes")


meltSummary <- meltSummary%>%
  arrange(Contrast)%>%
  mutate(Virus = c( rep("V186", 6), rep("SD90",6)),
         Time = as.factor(rep(c(rep("3",2),rep("8",2),rep("24",2)),2)))

meltSummary$Direction <- factor( meltSummary$Direction, levels = c("up","down"))

         
meltSummary$Time <- factor( meltSummary$Time, levels = c("3","8","24"))

library(ggplot2)
ggplot(meltSummary,aes(x=Time, y = Probes))+
  geom_point(aes(color = Virus))+
  facet_wrap(~Direction)+
  geom_line(aes(group=Virus, color=Virus))+
  theme(aspect.ratio = 1)+
  ggtitle("Number of DE probes (sample 3E excluded)")
```


V186 vs mock at 3hr

```{r,echo=FALSE,message=FALSE, warning=FALSE}

#adj p values <0.05 and lfc >=0.5

CELLStt1<-topTable(fit2,coef="V186.3vsMock.3", adjust.method = "BH", number=Inf, p.value=0.05, lfc=0.5)

CELLSselected1<-rownames(CELLStt1) #selecting the probes of interest

#from the CELLSexpressedProbes.lumi batch, select just the samples #corresponding to the corresponding coefs (time=24, treatment not SD90)

CELLSesetSel1<-CELLSexpressedProbes.lumi[CELLSselected1,targets$Time=="3"& targets$Treatment !="SD90" ]

#To have informative names along the bottom of the heatmap,replace the true Sample Names(numbers)with their corresponding treatment.
#i.e. look at the rows in CELLSexpressedProbes.lumi that match the colnames in exprs and provide the value for Treatment in those rows. Set that as the new colnames.

colnames(exprs(CELLSesetSel1))<-pData(CELLSexpressedProbes.lumi)[colnames(CELLSesetSel1),"Treatment"]

heatmap(exprs(CELLSesetSel1))
```

V186 vs mock at 8hr

```{r,echo=FALSE,message=FALSE, warning=FALSE}
#repeat for coefs 2
CELLStt2<-topTable(fit2,coef="V186.8vsMock.8", adjust.method = "BH", number=Inf, p.value=0.05, lfc=0.5)

CELLSselected2<-rownames(CELLStt2) #selecting the probes of interest

CELLSesetSel2<-CELLSexpressedProbes.lumi[CELLSselected2,targets$Time=="8"& targets$Treatment !="SD90" ]

colnames(exprs(CELLSesetSel2))<-pData(CELLSexpressedProbes.lumi)[colnames(CELLSesetSel2),"Treatment"]


heatmap(exprs(CELLSesetSel2))
```

V186 vs mock at 24hr

```{r,echo=FALSE,message=FALSE, warning=FALSE}

#and coef3
CELLStt3<-topTable(fit2,coef="V186.24vsMock.24", adjust.method = "BH", number=Inf, p.value=0.05, lfc=0.5)

CELLSselected3<-rownames(CELLStt3) #selecting the probes of interest

CELLSesetSel3<-CELLSexpressedProbes.lumi[CELLSselected3,targets$Time=="24"& targets$Treatment !="SD90" ]

colnames(exprs(CELLSesetSel3))<-pData(CELLSexpressedProbes.lumi)[colnames(CELLSesetSel3),"Treatment"]

heatmap(exprs(CELLSesetSel3))
```

```{r,echo=FALSE,message=FALSE, warning=FALSE}

CELLSttList<-list(CELLStt1, CELLStt2,CELLStt3)
ProbeIDandSymbol<-fData(RAW.lumi)[,c(1,4,7)]
save(ProbeIDandSymbol, file="CELLSProbeIDandSymbol.Rda")

#function to get short versions of the tt's and include
#probeID, entrez and symbols

getInteresting <- function(df){
short<-df[,c("logFC","adj.P.Val")]
  short$ProbeID<-rownames(short)
  rownames(short)<- NULL
  
  merge(ProbeIDandSymbol,short, by = "ProbeID")
 
}


CELLSshortTTlist<-lapply(CELLSttList,FUN=getInteresting)

names(CELLSshortTTlist) <-c("CELLS_V186.3","CELLS_V186.8","CELLS_V186.24")

#functions to get the up and down regulated probes

getUP<-function(df){
  df%>%
    filter(logFC>0)%>%
    select(ENTREZ_GENE_ID, logFC, adj.P.Val)
}


getDOWN<-function(df){
  df%>%
    filter(logFC<0)%>%
    select(ENTREZ_GENE_ID, logFC, adj.P.Val)
}


#get upreg
CELLSV186UP<-lapply(CELLSshortTTlist,FUN=getUP)
#get downreg      
CELLSV186DOWN<-lapply(CELLSshortTTlist,FUN=getDOWN)


#mapply looping over both list itself and the list names to name the txt files

mapply(write.table,CELLSV186UP,file=paste(names(CELLSV186UP),"UP",".txt", sep=""),sep="\t",row.names=FALSE)

mapply(write.table,CELLSV186DOWN,file=paste(names(CELLSV186DOWN),"DOWN",".txt",sep=""),sep="\t",row=FALSE)



###reading in the InnateDB results of the GO analysis. Here I will filter for just the probes that were annotated with the GO CC term "extracellular space" OR "extracellular region"


geneFolder <- "J:/MacLabUsers/HLADIK SHARED/Projects/Herpes study/Herpes-Project-1/INNATE DB RESULTS/geneSpreadsheets/CELLS"

geneFiles <- list.files(geneFolder, pattern="all")

geneFilesPath<-paste(geneFolder, "/",geneFiles, sep="")


geneIDBdata <- lapply(geneFilesPath, read.delim, sep="\t", header=TRUE)

names(geneIDBdata)<-str_replace(geneFiles,pattern=".txt",replacement="")

#There are a lot of columns in the files and several have lots of entries separated by |

#I will just keep the entrez ID, symbol (called "name")and goTerms


shortgeneIDBdata<-lapply(geneIDBdata, FUN = select, entrez, name,fullname,goTerms)

shortgeneIDBdata<-lapply(shortgeneIDBdata, FUN=filter, goTerms = str_detect(goTerms, pattern = "GO:0005615")==TRUE |str_detect(goTerms, pattern = "GO:0005576")==TRUE)

#annoyingly, there are several rows in the entrez column that have multiple numbers separated by a comma. All correspond to the same gene name (symbol). I want to separate out those rows, melt them so that there is one entrez id per line and then rbind with the rest of the data.




noCommas<- function(df){
  if(all(str_detect(df$entrez,",")==FALSE)){
    
    rownames(df)<-NULL
    
    m<-df[,c("entrez","name","fullname")]
    
    return(m) ## This part is for the df's in the list that DON'T have any commas and should be left alone
  }
  
  x<-filter(df,str_detect(df$entrez,",")==TRUE)
  
  x$position<-rownames(x)
  
  rownames(x)<-NULL
  
  
  y<-melt(str_split(x$entrez, pattern=","))#if comma, split, then melt
  
  colnames(y)<-c("singleEntrez","position")#position indicates the original position the entrez was in
  
  z<-merge(y, x, by="position")#will get the names and full names to repeat
  
  a<-z[,c("singleEntrez","name","fullname")] #just get the columns I want
  
  names(a)[1]<-"entrez"
  #next step rbinds the fixed data with its corresp data that had no commas and leaves out the goTerms column
  b<-filter(df,str_detect(df$entrez,",")==FALSE)%>%
    select(-goTerms)%>%
    rbind(a)
  
  return(b)

}




#lapply over the list of dataframes
CELLSExtracellularGenesV186<-lapply(shortgeneIDBdata, FUN=noCommas)

save(CELLSExtracellularGenesV186, file="CELLSExtracellularGenesV186.Rdata")


